robot/web/web.go

```diff
@@ -839,11 +686,9 @@ func (svc *webService) initHTTPServer(listenerTCPAddr *net.TCPAddr, options webo
 }
 
 // Initialize multiplexer between http handlers.
-func (svc *webService) initMux(options weboptions.Options) (*goji.Mux, error) {
+func (svc *webService) initMux(options weboptions.Options) *goji.Mux {
 	mux := goji.NewMux()
-	if err := svc.installWeb(mux, svc.r, options); err != nil {
-		return nil, err
-	}
+	mux.Handle(pat.New("/"), http.NotFoundHandler())
```

Comment:
@Otterverse see here - not found for `/`. How do health checks work for the agent?

Reply:
It looks for a 2XX status code on whatever URL viam-server gives as it's "serving" line during startup. Ex:

```
11/5/2024, 12:35:22 PM info rdk web/web.go:599 serving url https://artoo-main.p55vcc4g4x.local.viam.cloud:8080 alt_url https://0.0.0.0:8080

11/5/2024, 12:35:22 PM info viam-agent viamserver/viamserver.go:172 healthcheck URLs: https://artoo-main.p55vcc4g4x.local.viam.cloud:8080 https://localhost:8080

11/5/2024, 12:35:22 PM info viam-agent viamserver/viamserver.go:173 viam-server started

```

Basically agent sees that first line and parses it, and reports WHAT it parsed in that second line. Then it queries that every minute, looking for a 2XX http response (it doesn't care about the response body.)

Reply:
Awesome! I added a 200 status and "healthy" message here: [596b486](https://github.com/viamrobotics/rdk/pull/4520/commits/596b4860221b75452c81bc2ae55a2d6270fe619f)

@Otterverse are you able to verify that this works if I generate an AppImage?

Reply:
Agent doesn't use appimages. Those are the "old" method. Just do a static binary `make server-static` and then you can test it yourself. You can copy the built file to your device (running agent) and set the pin_url for viam-server to file:///path/to/local/viam-server to let it run the new version. (You can also use a remote/public URL if you find that easier.)

Reply:
I can try the same now too... will clone your branch and give it a whirl.

Reply:
Okay, running a build from this branch, looks good!

```
2024-11-05T20:19:31.030Z	INFO	rdk	web/web.go:449	serving	{"url":"https://artoo-main.p55vcc4g4x.local.viam.cloud:8080","alt_url":"https://0.0.0.0:8080"}
2024-11-05T20:19:31.031Z	INFO	viam-agent	viamserver/viamserver.go:172	healthcheck URLs: https://artoo-main.p55vcc4g4x.local.viam.cloud:8080 https://localhost:8080

...

2024-11-05T20:20:34.390Z	DEBUG	viam-agent	viamserver/viamserver.go:290	healthcheck for viam-server is good
2024-11-05T20:20:34.391Z	DEBUG	viam-agent	agent/manager.go:233	Subsystem healthcheck succeeded for viam-server
```

Reply:
Sorry for the delay here - thank you for taking a look! I am also going to try going through these steps for my understanding. Once I've confirmed that things are working as expected I'll move forward with the base PR and put this one up for explicit review.

Reply:
Sweet! I was able to verify this as well and everything is behaving as expected.